src/pycram/yarp_utils/yarp_joint_state_updater.py

```diff
@@ -0,0 +1,324 @@
+        self.left_arm_joints_names = RobotDescription.current_robot_description.get_actuated_joint_names("left_arm")
+        self.head_joints_names = ["neck_pitch","neck_roll","neck_yaw","eyes_tilt","r_eye_pan_joint","l_eye_pan_joint"]
+
+        if self.torso_joints_names is None:
+            print("Error in joint names")
+            return False
+
+        if self.right_arm_joints_names is None:
+            print("Error in joint names")
+            return False
+
+        if self.left_arm_joints_names is None:
+            print("Error in joint names")
+            return False
+
+        if self.head_joints_names is None:
+            print("Error in joint names")
+            return False
+
+        print("Initialization complete")
+        return True
+
+    def getName(self, name):
+        return f'/{self.module_name}{name}'
+
+
+    def connect_ports(self):
+        # status ports
+        connection_status = yarp.NetworkBase_connect("/"+self.robot_name_yarp+"/torso/state:o", self.state_torso_port_name, "tcp")
+        if not connection_status:
+            print("state torso port couldn't connect")
+            return False
+
+        connection_status = yarp.NetworkBase_connect("/"+self.robot_name_yarp+"/right_arm/state:o",self.state_right_arm_port_name,  "tcp")
+        if not connection_status:
+            print("state right_arm port couldn't connect")
+            return False
+
+        connection_status = yarp.NetworkBase_connect("/"+self.robot_name_yarp+"/left_arm/state:o",self.state_left_arm_port_name , "tcp")
+        if not connection_status:
+            print("state left_arm port couldn't connect")
+            return False
+
+        connection_status = yarp.NetworkBase_connect("/" + self.robot_name_yarp + "/head/state:o",
+                                                     self.state_head_port_name, "tcp")
+        if not connection_status:
+            print("state head port couldn't connect")
+            return False
+
+        return True
+
+    def interruptModule(self):
+        print("stopping the module")
+        self.handle_port.interrupt()
+        self.state_torso_port.interrupt()
+        self.state_right_arm_port.interrupt()
+        self.state_left_arm_port.interrupt()
+        self.state_head_port.interrupt()
+
+        return True
+
+    def close(self):
+        self.handle_port.close()
+        self.state_torso_port.close()
+        self.state_right_arm_port.close()
+        self.state_left_arm_port.close()
+        self.state_head_port.close()
+
+        return True
+
+    def respond(self, command, reply):
+
+        # Is the command recognized
+
+        reply.clear()
+
+        if command.get(0).asString() == "quit":
+            reply.addString("quitting")
+            return False
+        else:
+            reply.addString("other request")
+
+        return True
+
+
+    def getPeriod(self):
```

Comment:
The convention for python methods is snake_case, please rename to "get_period"

Reply:
It is a defined method in the parent class. I can't change it. 
It is also written in this way because yarp is written in C++ with a binding in pythin.